pandas/core/reshape/merge.py

```diff
@@ -116,20 +116,20 @@ def _groupby_and_merge(
 
     # if we can groupby the rhs
     # then we can get vastly better perf
-    try:
 
-        # we will check & remove duplicates if indicated
-        if check_duplicates:
-            if on is None:
-                on = []
-            elif not isinstance(on, (list, tuple)):
-                on = [on]
-
-            if right.duplicated(by + on).any():
-                _right = right.drop_duplicates(by + on, keep="last")
-                # TODO: use overload to refine return type of drop_duplicates
-                assert _right is not None  # needed for mypy
-                right = _right
+    # we will check & remove duplicates if indicated
+    if check_duplicates:
+        if on is None:
+            on = []
+        elif not isinstance(on, (list, tuple)):
+            on = [on]
+
+        if right.duplicated(by + on).any():
+            _right = right.drop_duplicates(by + on, keep="last")
```

Comment:
no risk of raising in 127/128?

Reply:
It was a risk already, right? This just ensures that a KeyError isn't catched unintended.

Reply:
makes sense, thanks